src/libstd/io/stdio.rs

```diff
@@ -872,6 +872,92 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Locks the current handle and reads a line of input, returning a `String` containing
+/// the input.
+///
+/// If you need more explicit control over
+/// locking, see the [`Stdin::lock`] and [`Stdout::lock`] methods.
+///
+/// [`Stdin::lock`]: struct.Stdin.html#method.lock
+/// [`Stdout::lock`]: struct.Stdout.html#method.lock
+///
+/// ### Note: Safety and Usage
+/// This function is not the recommended solution to get user input, it should
+/// only be used for simple actions. If you require more explicit control over
+/// getting user input, see the [`Stdin::read_line`] method.
+///
+/// [`Stdin::read_line`]: struct.Stdin.html#method.read_line
+///
+/// # Examples
+///
+/// ```rust
+/// use std::io::input_prompt;
+///
+/// fn main() {
+///     let user_input = input_prompt("Please enter some text: ");
+///
+///     println!("You typed: {}", user_input);
+///
+/// }
+/// ```
+#[unstable(
+    feature = "input_prompt",
+    reason = "this function may be replaced with a more general mechanism",
+    issue = "none"
+)]
+pub fn input_prompt(prompt: &str) -> String {
+    stdout()
+        .write(prompt.as_bytes())
+        .unwrap();
+    return input();
```

Comment:
```suggestion
    stdout().write(prompt.as_bytes()).unwrap();
    input()
```

Should we even unwrap there?

Reply:
`write()` returns a `Result<usize>`, which might be worth handling in case something goes wrong. Also gives a warning for not using `unwrap()`

Reply:
Ah, I meant to use `expect`. Not sure if that's better.